### ISSUE TITLE
DeferredQuery single F[T] return type

### ISSUE THREAD

**Issue by geoffjohn11** [neotypes/neotypes#62]
DeferredQuery is defined as: 
```def single[F[_]](session: Session[F])(implicit F: Async[F], rm: ResultMapper[T]): F[T]```

However, issues might arise when the neo4j datastore doesn't have a corresponding value.  In that case I imagine a null will be returned.

I propose a change to single to reflect the possibility that the value might not exist:

```def single[F[_]](session: Session[F])(implicit F: Async[F], rm: ResultMapper[T]): F[Option[T]]```

**Comment by BalmungSan** [comment]
@geoffjohn11 It won't return `null` but rather the effect **F** will be completed with an _exception_, which you can later catch or let it crash.

Now, we could discuss whenever we could make all executions return `Either[NeotypesException, T]` instead of just `T`.

IMHO, this is a bad idea, because this doesn't add any value _(one can always transform any `F[T]` into some `F[Either[Exeption, T]`)_ and makes more difficult for people that do not want to handle such errors _(yet?)_. For example, for all my use cases, I know the format of my data, so that **either** will always be a **right** and in case of it being a **left** I really can't do anything with it, I would just rethrow it so my app crashes.

Also, if you really want just an `Opion[T]` then you can simply:

```scala
"query".query[Option[T]].single(s)
```

So, in conclusion, I believe our current model is flexible enough to support any use case.
